backend/src/datasources/userDatasource.js

```diff
@@ -56,6 +74,8 @@ export class UserDataSource extends DataSource {
 
```

Comment:
`removeUser` misses the ` await this.getAll()` call. Also the removal has no impact on the data in your database. It looks like no user is actually removed from the database.